azkaban-common/src/main/java/azkaban/database/AzkabanDatabaseSetup.java

```diff
@@ -255,30 +261,29 @@ private void findOutOfDateTables() {
       return null;
     }
 
-    String updateFileNameVersion = UPDATE_SCRIPT_PREFIX + table + "." + version;
+    String updateFileNameVersion = UPDATE_SCRIPT_PREFIX + table + "." + currentVersion;
     for (File file : createScripts) {
       String fileName = file.getName();
       if (fileName.compareTo(updateFileNameVersion) > 0) {
-        if (fileName.startsWith(updateFileNameVersion)) {
-          continue;
-        }
-
         String[] split = fileName.split("\\.");
-        String versionNum = "";
+        String updateScriptVersion = "";
 
         for (int i = 2; i < split.length - 1; ++i) {
           try {
             Integer.parseInt(split[i]);
-            versionNum += split[i] + ".";
+            updateScriptVersion += split[i] + ".";
           } catch (NumberFormatException e) {
             break;
           }
         }
-        if (versionNum.endsWith(".")) {
-          versionNum = versionNum.substring(0, versionNum.length() - 1);
-
-          if (versionNum.compareTo(version) == 0) {
-            versions.add(versionNum);
+        if (updateScriptVersion.endsWith(".")) {
+          updateScriptVersion = updateScriptVersion.substring(0, updateScriptVersion.length() - 1);
+
+          // add to update list if updateScript will update above current
+          // version and upto targetVersion in database.properties
+          if (updateScriptVersion.compareTo(currentVersion) > 0
+            && updateScriptVersion.compareTo(this.version) <= 0) {
```

Comment:
You may want to state somewhere that padding must be enforced in the major/minor versions reach multi digits. otherwise this string comparison may fail.